c7n/resources/appmesh.py

```diff
@@ -174,23 +212,42 @@ class resource_type(TypeInfo):
         service = 'appmesh'
 
         # https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsappmesh.html#awsappmesh-resources-for-iam-policies  # noqa
+        # TODO IS THIS EVEN USED AT RUNTIME?
+        # Junk value doesn't seem to affect the functionality of the extension
+        # but does fail the test PolicyMetaLint.test_valid_arn_type
+        # but that isn't a functional test.
         arn_type = "mesh"
 
         # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appmesh-virtualgateway.html  # noqa
-        cfn_type = config_type = 'AWS::AppMesh::VirtualGateway'
+        # TODO IS THIS EVEN USED AT RUNTIME?
+        # Junk value doesn't seem to affect the functionality of the extension
+        # but does fail the test PolicyMetaLint.test_cfn_resource_validity
+        # but that isn't a functional test.
+        cfn_type = 'AWS::AppMesh::VirtualGateway'
+
+        # TODO IS THIS EVEN USED AT RUNTIME?
+        # Junk value doesn't seem to affect the functionality of the extension
+        # but does fail the test PolicyMetaLint.test_cfn_resource_validity
+        # but that isn't a functional test.
+        config_type = 'AWS::AppMesh::VirtualGateway'
 
         # if a resource type is supported by resource group tagging
         # api setting this value get tag filters/actions
         universal_taggable = object()
 
-        # id: Path to "id" field in the
+        # id: Path to "id" field in the .... IN THE WHAT???
+        # TODO: WHERE IS THIS EVEN USED AT RUNTIME?
         id = 'meshName'
 
         # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appmesh-virtualgateway.html  # noqa
-        arn = "metadata.arn"
+        # TODO: WHERE IS THIS EVEN USED AT RUNTIME?
+        arn = "metadata.arn!!!!!!!"

```

Comment:
[`get_arns()`](https://github.com/cloud-custodian/cloud-custodian/blob/503d73d89d3b97e94d97b2c8bdc164d0acac40e8/c7n/query.py#L617-L635) pops up in a few places (like [building a set of resource ARNs for bulk tagging operations](https://github.com/cloud-custodian/cloud-custodian/blob/503d73d89d3b97e94d97b2c8bdc164d0acac40e8/c7n/tags.py#L89-L90)).

That method tries to get an ARN for a resource type _somehow_, which means:
- Do we already have an `arn` from the list/describe response? Cool, [use that](https://github.com/cloud-custodian/cloud-custodian/blob/503d73d89d3b97e94d97b2c8bdc164d0acac40e8/c7n/query.py#L630)
- Do we have an `arn_type` defined? Cool, we can use that to [build an ARN](https://github.com/cloud-custodian/cloud-custodian/blob/503d73d89d3b97e94d97b2c8bdc164d0acac40e8/c7n/query.py#L642-L648) from the parts we have

Reply:
for AppmeshVirtualGateway this wont work anyway as "arn" cannot be a path (no dots) and the ARN isn't top level in the datastruture.

Reply:
overriding get_arns means the metadata for arn isn't used.

Reply:
yep - got that one thanks